File: src/main/java/tom/com/student/Student.java
package tom.com.student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest() {
//方法二        int max = (english>math)?english:math;
//  /方法一      if (english>math){
//            max=english;
//        }else{
//            max=math;
        //}
        return (english > math) ? english : math;
    }

    public int getAverage() {
        return (english + math) / 2;
    }

    public void print() {
        int average = getAverage();
        System.out.println(name + " " + english + " " + math + " " + getAverage()
                + ((getAverage() >= 60) ? " pass" : " failed"));
//        if (getAverage()>=60){
//            System.out.println("\tpass");
//        }else {
//            System.out.println("\tfailed");
        char grading = 'F';
        switch (average/10){
            case 10:
            case 9:
                grading ='A';
                break;
            case 8:
                grading ='B';
                break;
            case 7:
                grading ='C';
                break;
            case 6:
                grading ='D';
                break;
                default:
                    grading='F';


        }
//        if (average >= 90 && average < 100) {
//            grading = 'A';
//        } else if (average >= 80 && average < 90) {
//            grading = 'B';
//        } else if (average >= 70 && average < 80) {
//            grading ='C';
//        }else if (average>=60 && average<70){
//            grading ='D';
//        }else {
//            grading ='F';
//        }
        System.out.println(" "+grading);
    }
}

File: src/main/kotlin/com/tom/Student.kt
package com.tom

import java.util.*

fun main(args: Array<String>) {
    val stu = Student("Jack", 66, 99)
    stu.print()
    println("highest score ${stu.highest()}")

    //userInput()
}

class Student(var name:String, var english : Int, var math :Int){
    fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")

    }
    fun getAverage() = (english+math)/2

    fun passOrFailed() = if (getAverage()>=60)"\tpass" else "\tfailed"

    fun highest() = if (english>math) english else math

    fun grading() = when(getAverage()){
            in 90..100 ->'A'
            in 80..89 ->'B'
            in 70..79 ->'C'
            in 60..69 ->'D'
            else -> 'F'
        }

//   方法二 fun grading() : Char{
//        return when(getAverage()){
//            in 90..100 ->'A'
//            in 80..89 ->'B'
//            in 70..79 ->'C'
//            in 60..69 ->'D'
//            else -> 'F'

//   方法一     var grading = when(getAverage()){
//            in 90..100 ->'A'
//            in 80..89 ->'B'
//            in 70..79 ->'C'
//            in 60..69 ->'D'
//            else -> 'F'
//        }
//        return grading



//      if(english>math){
        //print("english")
//        max = english
//        }else{
//           //print("math")
//            max = math
//        }


private fun userInput() {
    val scanner = Scanner(System.`in`)
    print("Please Enter you name:")
    var name = scanner.next()
    print("Please Enter you english:")
    var english = scanner.nextInt()
    print("plase enter your math score:")
    var math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
}}


